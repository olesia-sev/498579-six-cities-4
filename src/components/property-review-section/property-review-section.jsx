import React, {useEffect, useState} from 'react';
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {ReviewsList} from "../reviews-list/reviews-list";
import PropertyReviewForm from "../property-review-form/property-review-form";
import {getReviews} from "../../reducer/data/selectors";
import {Operation as DataOperation} from "../../reducer/data/data";
import {withRouter} from "react-router-dom";
import {reviewTypeArray} from "../../prop-types/prop-types";

const PropertyReviewSection = ({reviews, authStatus, loadReviews}) => {
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    loadReviews().then(() => {
      setIsLoading(false);
    });
  }, []);

  if (isLoading) {
    return null;
  }

  return (
    <section className="property__reviews reviews">
      <h2 className="reviews__title">Reviews &middot; <span className="reviews__amount">{reviews.length}</span></h2>

      <ReviewsList reviews={reviews} />

      {authStatus === `AUTH` ? <PropertyReviewForm /> : null}

    </section>
  );
};

PropertyReviewSection.propTypes = {
  authStatus: PropTypes.string.isRequired,
  reviews: reviewTypeArray,
  loadReviews: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => {
  return {
    reviews: getReviews(state)
  };
};

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    loadReviews: () => dispatch(DataOperation.loadReviews(ownProps.match.params.id))
  };
};

const ConnectedPropertyReviewSection = withRouter(connect(mapStateToProps, mapDispatchToProps)(PropertyReviewSection));

export {ConnectedPropertyReviewSection as PropertyReviewSection};
