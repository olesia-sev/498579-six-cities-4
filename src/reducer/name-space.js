const NameSpace = {
  DATA: `DATA`,
  APP: `APP`,
  USER: `USER`,
};

export {NameSpace};
