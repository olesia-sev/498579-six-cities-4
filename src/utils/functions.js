export const getRandomNumberInRange = (min, max) => {
  return Math.floor(Math.random() * (max - min + 1)) + min;
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};
