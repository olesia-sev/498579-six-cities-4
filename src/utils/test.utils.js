export const optionsAmount = 100;

export const testString = `Test string`;

export const emptyFunction = () => {};

export const cardDataArray = [
  {
    id: 10,
    img: `https://placedog.net/260/200?id=2`,
    images: [
      `https://placedog.net/260/200?id=3`,
      `https://placedog.net/260/200?id=33`,
      `https://placedog.net/260/200?id=12`,
      `https://placedog.net/260/200?id=10`,
      `https://placedog.net/260/200?id=11`,
      `https://placedog.net/260/200?id=15`,
    ],
    price: 150,
    title: `Title`,
    placeType: `Flat`,
    numberOfBedrooms: 2,
    maxGuests: 3,
    rating: 4,
    saved: true,
    premium: true,
    featuresInside: [
      `Wi-Fi`, `Heating`,
    ],
    hostName: `Pete`,
    hostAvatar: `https://placedog.net/260/200?id=5`,
    userPro: true,
    description: `Description`,
  },
  {
    id: 100,
    img: `https://placedog.net/260/200?id=3`,
    images: [
      `https://placedog.net/260/200?id=33`,
      `https://placedog.net/260/200?id=99`,
      `https://placedog.net/260/200?id=89`,
      `https://placedog.net/260/200?id=29`,
      `https://placedog.net/260/200?id=13`,
      `https://placedog.net/260/200?id=16`,
    ],
    price: 150,
    title: `Title test`,
    placeType: `Flat`,
    numberOfBedrooms: 1,
    maxGuests: 2,
    rating: 4,
    saved: false,
    premium: false,
    featuresInside: [
      `Wi-Fi`, `Heating`,
    ],
    hostName: `Pete`,
    hostAvatar: `https://placedog.net/260/200?id=57`,
    userPro: false,
    description: `Description`,
  }
];
