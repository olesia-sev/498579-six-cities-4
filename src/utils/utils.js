export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const SORT_TYPES = {
  POPULAR: `Popular`,
  TO_HIGH_PRICE: `Price: low to high`,
  TO_LOW_PRICE: `Price: high to low`,
  TOP_RATED: `Top rated first`,
};

export const getSortedOffers = (offers, activeSort) => {
  switch (activeSort) {
    case SORT_TYPES.POPULAR:
      return offers.slice();
    case SORT_TYPES.TO_HIGH_PRICE:
      return offers.slice().sort((a, b) => a.price - b.price);
    case SORT_TYPES.TO_LOW_PRICE:
      return offers.slice().sort((a, b) => b.price - a.price);
    case SORT_TYPES.TOP_RATED:
      return offers.slice().sort((a, b) => b.rating - a.rating);
    default:
      return offers.slice();
  }
};

export const MAX_REVIEWS_LENGTH = 10;

export const AppRoute = {
  LOGIN: `/login`,
  ROOT: `/`,
  FAVOURITES: `/favorites`,
  OFFER_DETAIL: `/offers/:id`,
};

export const RATING_STAR_WIDTH = 20;

export const MAX_IMAGES_AMOUNT = 6;
export const SLICE_BEGIN = 0;
