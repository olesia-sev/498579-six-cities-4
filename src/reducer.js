import {extend} from './utils/functions';
import cities from "./mocks/cities";
import offers from "./mocks/offers";

export const initialState = {
  activeCityId: 10,
  cities,
  offers,
};

const ActionType = {
  SET_ACTIVE_CITY: `SET_ACTIVE_CITY`,
};

const ActionCreator = {
  setActiveCity: (id) => ({
    type: ActionType.SET_ACTIVE_CITY,
    payload: id,
  })
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SET_ACTIVE_CITY:
      return extend(state, {
        activeCityId: action.payload,
      });

    default: {
      return state;
    }
  }
};

export {reducer, ActionType, ActionCreator};
