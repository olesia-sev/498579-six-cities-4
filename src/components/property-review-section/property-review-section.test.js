import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter as Router} from "react-router-dom";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {PropertyReviewSection} from "./property-review-section";
import {authStatus, reviews} from "../../utils/test.utils";
import {NameSpace} from "../../reducer/name-space";

const mockStore = configureStore([]);

it(`Property Review Section should be rendered`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      reviews,
    },
  });

  const tree = renderer
    .create(<Router>
      <Provider store={store}>
        <PropertyReviewSection authStatus={authStatus} />
      </Provider>
    </Router>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
