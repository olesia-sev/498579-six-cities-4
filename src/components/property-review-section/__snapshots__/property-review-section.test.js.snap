// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property Review Section should be rendered 1`] = `null`;
